### PR TITLE
feat: replace mailto contact handoff with Resend-powered server form (#96)

### LINKED ISSUE #96: Replace mailto contact handoff with a real contact form
## Summary

Replace the current contact-form behavior, which pre-populates the visitor's local email client via `mailto:`, with a true server-side contact form that submits directly from the Sea Saba website.

## Current behavior

`components/contact-form.tsx` currently builds a `mailto:info@seasaba.com` draft and hands off to the visitor's email client. The site does not currently send or store contact submissions server-side.

## Goal

Allow a visitor to complete and submit the contact form without leaving the website or opening a local mail client.

## Requirements

- Preserve the existing contact-form fields and UX unless a change is justified.
- Submit the form to a server-side endpoint or server action.
- Send the inquiry to the appropriate Sea Saba inbox using the existing Vercel-managed Resend integration and verified sending domain `mail.seasaba.com`.
- Keep WhatsApp as an alternate contact method.
- Provide clear success, validation, and failure states in the UI.
- Prevent duplicate submissions and basic abuse/spam where practical.
- Do not expose `RESEND_API_KEY` or any server-only credential to the client.
- Preserve existing analytics behavior and add/update submit-success/failure tracking only if consistent with the current analytics model.
- Add tests for successful submission, validation failures, provider failure, and duplicate/double-submit behavior.
- Update documentation that currently states the website does not send/store messages.

## Design considerations

- Prefer a small, explicit server-side implementation over adding unnecessary infrastructure.
- Decide whether submissions should only be emailed or also persisted. If persistence is proposed, document the reason and privacy implications before implementing it.
- Use an appropriate sender on `mail.seasaba.com`; replies should route to the visitor's supplied email address where supported.
- Keep failure handling user-friendly and avoid losing the visitor's entered message if sending fails.

## Acceptance criteria

- Visitor can submit the form entirely on-site.
- Sea Saba receives the inquiry by email.
- The visitor sees a clear success state after a confirmed send.
- Resend/provider failure does not falsely report success.
- No secret values are committed or exposed client-side.
- Relevant unit/integration/E2E tests pass.
- Documentation accurately describes the new contact flow.

### PR DESCRIPTION
**Pull request by spizeck**
Closes #96

## Summary

Replaces the `mailto:` contact handoff with a server-side submission. A visitor completes the existing form, clicks Send, and the inquiry is delivered to Sea Saba by email — no local mail client, no redirect. WhatsApp remains a fully client-side alternate path.

## Previous architecture

`components/contact-form.tsx` built a `mailto:info@seasaba.com` URL with a prefilled subject/body and called `window.location.href` — delivery depended on the visitor having a configured mail client.

## New architecture

- **`POST /api/contact`** (`app/api/contact/route.ts`) — thin route handler: body-size bound (32 KB), JSON parse, honeypot, validation, best-effort rate limit, provider call.
- **`lib/contact.ts`** — pure validation + email construction (no Next/provider imports), unit-tested in isolation.
- **Form** posts JSON, then shows an in-place success panel (focus moved to a `role="status"` region) or a `role="alert"` error with fields preserved for retry.

## Email behavior

- **From:** `Sea Saba Website <website@${RESEND_EMAIL_DOMAIN}>` → `website@mail.seasaba.com` on the verified domain
- **To:** `CONTACT.email` (`info@seasaba.com`) — canonical inbox from `lib/constants.ts`
- **Reply-To:** the visitor's submitted address — staff hit Reply; we never send "from" the visitor
- **Subject:** `<canonical inquiry subject> — <visitor name>`; inquiry context comes from `INQUIRY_TYPES`, and name is newline-stripped so no header/subject injection is possible
- **Body:** plain text + simple HTML table (name, email, inquiry label, WhatsApp/dates/students/certification/logged dives when present, preferred contact, message). All visitor content is HTML-escaped; no IP, fingerprint, or tracking data included.

## Validation

Server-side (`lib/contact.ts`): name, email (format + ≤254), inquiry type (must resolve via `inquiryFor` — arbitrary values rejected), message; all optional fields length-limited; CR/LF stripped from single-line fields. Client validation kept for immediate feedback; server errors map back onto the same field keys and are surfaced in place.

## Anti-spam measures

- Honeypot field (off-screen, `tabIndex={-1}`): filled → `200 {ok:true}` accept-and-drop, never sent
- Per-IP in-memory rate limit (5 / 10 min)
- Body-size bound + strict field limits
- No CAPTCHA — proportionate for a small-business form

## Duplicate-send strategy and limits

The client generates a `submissionId` (UUID) per distinct payload and sends it on every attempt; the server forwards it as Resend's `idempotencyKey`, so retries/double-clicks/network replays of the *same* content are deduplicated by the provider (~24 h window). Editing any field after a failure generates a fresh key so new content is never suppressed.

**Not provided:** durable cross-instance dedupe on our side (the in-memory rate limiter is per-instance and documented as best-effort). Distributed limiting/dedupe would need a shared store (e.g. Vercel KV) — deliberately out of scope.

## Failure behavior

Provider rejection or throw → `502` with a generic customer-safe message; the UI keeps all entered fields, shows the error with WhatsApp and the canonical `mailto:` address as manual fallbacks, and allows retry. No automatic mailto fallback. Missing `RESEND_*` config → `500` + generic message; secrets never reach the client.

## Analytics

- `contact_form_submit` now fires only on confirmed server acceptance (`method: "email"`) or a WhatsApp handoff (`method: "whatsapp"`) — previously it meant "mailto handoff prepared"
- New `contact_form_error` on send failure — no PII in either event
- Documented in `docs/ANALYTICS_SEO.md`

## Privacy / docs

- `/privacy`: email-processor row updated to name the Resend server-side flow; stale Rezdy references corrected to Checkfront
- `.env.example`: `RESEND_API_KEY`, `RESEND_EMAIL_DOMAIN` documented as **server-only** (section header now distinguishes public vs server vars)
- `docs/OPERATIONS.md` + `docs/TESTING.md` updated for the new architecture and coverage
- CSP unchanged — the endpoint is same-origin (`connect-src 'self'` covers it); no new browser connections

## Test coverage (all mocked — no real email ever sent)

- **Unit (24):** valid submission, missing/malformed fields, unknown inquiry type, over-limit fields, honeypot accept-and-drop, rate limit, provider reject/throw, missing config, subject safety, Reply-To, HTML escaping, no secret leakage
- **Integration (14):** submit → success state, field preservation + retry on failure, double-click protection, `?interest=` preselection, WhatsApp path
- **E2E:** submit against intercepted endpoint → in-place confirmation + `contact_form_submit`; mocked 502 → error state, preserved fields, `contact_form_error`; keyboard submit; double-submit = one request
- **Production smoke:** new assertion that `POST /api/contact` with invalid data returns `400` (fails before Resend — provably sends nothing)

## Environment requirements

Vercel already has `RESEND_API_KEY` + `RESEND_EMAIL_DOMAIN` (`mail.seasaba.com` verified). No client-side exposure — neither var is `NEXT_PUBLIC_*`. Without them the endpoint returns a safe 500 and the UI offers WhatsApp/email fallback.

## Validation results

- `npm run check` (docs/env/hygiene), lint, typecheck ✅
- Unit/integration: 144 tests, all pass ✅
- `build:test` ✅; full Playwright: **156 pass / 57 skips / 0 fail** ✅
- `test:perf`: all budgets met (`/dive-log` LCP warn only — pre-existing) ✅
- Production spec suite against a local build: 17 pass / 1 prod-only skip ✅
- `npm audit`: 0 vulnerabilities; `resend` pinned at 6.27.0 (>7 days old)

## Manual production verification (safe)

After deploy: submit the form with a real-but-own test address once to confirm delivery to `info@seasaba.com`, or ask the owner to. Everything else is verified without sending mail.

## Remaining owner decisions

None.

Generated with [Devin](https://devin.ai)

## Summary by Sourcery

Replace the mailto contact handoff with a validated, resilient server-side email submission flow powered by Resend.

New Features:
- Deliver contact-form inquiries through a server-side Resend email endpoint with visitor Reply-To handling and in-place confirmation.
- Retain WhatsApp as an alternate client-side contact path.

Bug Fixes:
- Replace mail-client-dependent mailto handoffs with reliable server-side delivery and safe retry behavior.

Enhancements:
- Add server-side validation, honeypot protection, request limits, best-effort rate limiting, provider-safe failures, and idempotent retries.
- Improve contact-form accessibility and error recovery while preserving entered fields.
- Update analytics to distinguish confirmed submissions from send failures.
- Correct privacy and operational references for Resend and Checkfront.

Build:
- Add the pinned Resend dependency and document server-only email configuration.

Documentation:
- Update analytics, privacy, operations, and testing documentation for the server-side contact flow.

Tests:
- Add unit, integration, end-to-end, and production coverage for validation, email construction, delivery outcomes, retries, anti-spam controls, and safe failure behavior.

### CODE REVIEW

File: app/api/contact/route.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  try {
+    payload = await request.json();
+  } catch {
+    return json({ ok: false, error: GENERIC_FAILURE }, 400);
+  }
+
+  // Honeypot: the field is invisible and unfocusable for humans; a filled
+  // value means an automated submitter. Accept-and-drop — never send.
+  if (typeof (payload as Record<string, unknown>).website === "string" &&
+      (payload as Record<string, unknown>).website !== "") {
```
[the remark this answers is not in the document]
Comment (spizeck) [reply app/api/contact/route.ts]: Fixed in 61b4582. The honeypot check now guards `payload !== null && typeof payload === "object"` before reading `.website`, so a literal `null` JSON body falls through to `validateContactSubmission` and returns the documented 400 field-error response. Regression test added: "rejects a JSON null body with a validation error, not a crash" in tests/unit/contact.test.ts.

File: app/api/contact/route.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+export async function POST(request: Request) {
+  const length = Number(request.headers.get("content-length") ?? 0);
+  if (length > MAX_BODY_BYTES) return json({ ok: false, error: GENERIC_FAILURE }, 413);
+
+  let payload: unknown;
+  try {
+    payload = await request.json();
+  } catch {
```
[the remark this answers is not in the document]
Comment (spizeck) [reply app/api/contact/route.ts]: Fixed in 61b4582. Content-Length is now only a fast-path short-circuit; the real bound is enforced in `readJsonBody`, which streams `request.body` with a reader, cancels, and returns 413 as soon as cumulative bytes exceed 32 KB — chunked bodies without a trustworthy header can no longer be fully buffered before rejection. Regression test added: "bounds streamed bodies that carry no Content-Length" (ReadableStream body, no Content-Length) in tests/unit/contact.test.ts.

File: app/api/contact/route.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    return true;
+  }
+  recent.push(now);
+  hits.set(key, recent);
+  // Bound the map so a spray of unique keys can't grow it without limit.
+  if (hits.size > 5000) {
+    for (const [k, times] of hits) {
+      if (times.every((t) => now - t >= RATE_WINDOW_MS)) hits.delete(k);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (spizeck) [reply app/api/contact/route.ts]: Fixed in 61b4582. After expired-key cleanup, the limiter now evicts oldest (insertion-ordered) entries until `hits.size` is back under the 5,000-key cap, so a spray of distinct fresh source IPs cannot grow the map without bound. This intentionally resets the oldest keys' windows under a spray — an accepted trade-off for a best-effort per-instance brake, documented in the code comment. Regression test added: "evicts the oldest tracked keys when the rate-limit map hits its cap" plus a basic "rate-limits repeated submissions from one source IP" test in tests/unit/contact.test.ts.
